Major changes to OS Drive expanding approach

bahreex opened the pull request:
With the introduction of Managed Disks in Azure, the approach for disk resizing has changed. The way to expand Managed disk size is different from Unmanaged disks. What approach and PS code is shown in this docs article is only for expanding OS drive size for Unmanaged disks. I have added the PS approach and code for expanding OS drives size for Managed disks. I have tried to keep the flow and formatting intact as per the original, and have also added the relevant section on Data Disks pertaining to Managed disks too. There were some script errors in existing PS code and some missing Info, which I corrected.

azurecla commented:
Hi __@mkarbarz__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla.azure.com.
    </span>

TTYL, AZPRBOT;

azurecla commented:
__@mkarbarz__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, AZPRBOT;

deneha commented:
@mkarbarz When you have completed your test, please close this pull request.  Thank you.

bahreex committed:
Major changes to OS Drive expanding approach

With the introduction of Managed Disks in Azure, the approach for disk resizing has changed. The way to expand Managed disk size is different from Unmanaged disks. What approach and PS code is shown in this docs article is only for expanding OS drive size for Unmanaged disks. I have added the PS approach and code for expanding OS drives size for Managed disks. I have tried to keep the flow and formatting intact as per the original, and have also added the relevant section on Data Disks pertaining to Managed disks too. There were some script errors in existing PS code and some missing Info, which I corrected.

kirpasingh reviewed (approved):
Thank you @bahreex, changes look good to me.
